diff -Nur pam-pld-0.75.6.orig/libpam/Makefile.am pam-pld-0.75.6/libpam/Makefile.am
--- pam-pld-0.75.6.orig/libpam/Makefile.am	Sun Jun  3 00:54:01 2001
+++ pam-pld-0.75.6/libpam/Makefile.am	Thu Jun 13 14:13:07 2002
@@ -7,9 +7,9 @@
 DEFS			= -DDEFAULT_MODULE_PATH=\""/lib/security/"\" \
 			  $(STATIC) $(DYNAMIC)
 
-libexecdir		= @prefix@/lib
+libdir			= @prefix@/lib
 
-libexec_LTLIBRARIES	= libpam.la
+lib_LTLIBRARIES		= libpam.la
 
 CLEANFILES		= *~
 
diff -Nur pam-pld-0.75.6.orig/libpam_misc/Makefile.am pam-pld-0.75.6/libpam_misc/Makefile.am
--- pam-pld-0.75.6.orig/libpam_misc/Makefile.am	Sun Mar  3 00:55:41 2002
+++ pam-pld-0.75.6/libpam_misc/Makefile.am	Thu Jun 13 14:13:07 2002
@@ -1,10 +1,10 @@
 SUBDIRS			= include
 
-libexecdir		= @prefix@/lib
+libdir			= @prefix@/lib
 
 LIBS			=
 
-libexec_LTLIBRARIES	= libpam_misc.la
+lib_LTLIBRARIES		= libpam_misc.la
 
 INCLUDES		= -I$(top_srcdir) -I$(top_srcdir)/include \
 			  -I$(srcdir)/include/security
diff -Nur pam-pld-0.75.6.orig/libpamc/Makefile.am pam-pld-0.75.6/libpamc/Makefile.am
--- pam-pld-0.75.6.orig/libpamc/Makefile.am	Sun Jun  3 00:54:16 2001
+++ pam-pld-0.75.6/libpamc/Makefile.am	Thu Jun 13 14:13:07 2002
@@ -17,11 +17,11 @@
 DEFS			= -DDEFAULT_MODULE_PATH=\""/lib/security"\" \
 			  $(STATIC) $(DYNAMIC)
 
-libexecdir		= @prefix@/lib
+libdir			= @prefix@/lib
 
 CLEANFILES		= *~
 
-libexec_LTLIBRARIES	= libpamc.la
+lib_LTLIBRARIES		= libpamc.la
 
 libpamc_la_SOURCES	= pamc_client.c pamc_converse.c pamc_load.c
 
diff -Nur pam-pld-0.75.6.orig/modules/pam_access/Makefile.am pam-pld-0.75.6/modules/pam_access/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_access/Makefile.am	Fri Mar  1 01:46:21 2002
+++ pam-pld-0.75.6/modules/pam_access/Makefile.am	Thu Jun 13 14:13:15 2002
@@ -8,7 +8,7 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 access_confdir		= /etc/security
 access_conf_DATA	= access.conf
@@ -17,7 +17,7 @@
 			  -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_access.la
+pkglib_LTLIBRARIES	= pam_access.la
 endif
 
 pam_access_la_SOURCES	= pam_access.c
@@ -36,7 +36,6 @@
 			  -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_access.lo
-libpam_access_lo_SOURCES = pam_access.c
 libpam_access.lo$(EXEEXT): pam_access.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_cap/Makefile.am pam-pld-0.75.6/modules/pam_cap/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_cap/Makefile.am	Fri Mar  1 01:46:21 2002
+++ pam-pld-0.75.6/modules/pam_cap/Makefile.am	Thu Jun 13 14:13:43 2002
@@ -7,7 +7,7 @@
 if CAPLIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_cap.la
+pkglib_LTLIBRARIES	= pam_cap.la
 endif
 
 pam_cap_la_SOURCES	= pam_cap.c
@@ -22,7 +22,7 @@
 
 LIBS			= -lcap
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 cap_confdir		= /etc/security
 cap_conf_DATA		= capability.conf
@@ -44,7 +44,6 @@
 
 if CAPLIB_PRESENT
 noinst_PROGRAMS		= libpam_cap.lo
-libpam_cap_lo_SOURCES	= pam_cap.c
 endif
 
 STATIC_INCLUDES		= -DCONFILE=\"$(cap_confdir)/$(cap_conf_DATA)\" \
diff -Nur pam-pld-0.75.6.orig/modules/pam_console/Makefile.am pam-pld-0.75.6/modules/pam_console/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_console/Makefile.am	Fri Mar  1 01:46:21 2002
+++ pam-pld-0.75.6/modules/pam_console/Makefile.am	Thu Jun 13 14:13:49 2002
@@ -8,7 +8,7 @@
 
 LIBS			= -lc
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 console_confdir		= /etc/security
 console_conf_DATA	= consoles
@@ -18,7 +18,7 @@
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC) -D_GNU_SOURCE
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_console.la
+pkglib_LTLIBRARIES	= pam_console.la
 endif
 
 pam_console_la_SOURCES	= pam_console.c
@@ -37,7 +37,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC) -D_GNU_SOURCE
 			   
 noinst_PROGRAMS		= libpam_console.lo
-libpam_console_lo_SOURCES = pam_console.c
 libpam_console.lo$(EXEEXT): pam_console.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_cracklib/Makefile.am pam-pld-0.75.6/modules/pam_cracklib/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_cracklib/Makefile.am	Fri Mar  1 01:46:21 2002
+++ pam-pld-0.75.6/modules/pam_cracklib/Makefile.am	Thu Jun 13 14:13:56 2002
@@ -7,7 +7,7 @@
 if CRACKLIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_cracklib.la
+pkglib_LTLIBRARIES	= pam_cracklib.la
 endif
 
 pam_cracklib_la_SOURCES	= pam_cracklib.c
@@ -24,7 +24,7 @@
 pam_cracklib_la_LIBADD	= -lcrack
 endif
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -DCRACKLIB_DICTPATH=\"$(CRACKLIB_DICTPATH)\" \
 			  -I$(top_srcdir)/include $(DYNAMIC)
@@ -42,7 +42,6 @@
 if CRACKLIB_PRESENT
 
 noinst_PROGRAMS		= libpam_cracklib.lo
-libpam_cracklib_lo_SOURCES = pam_cracklib.c
 
 endif
 
diff -Nur pam-pld-0.75.6.orig/modules/pam_debug/Makefile.am pam-pld-0.75.6/modules/pam_debug/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_debug/Makefile.am	Wed Mar  6 23:55:27 2002
+++ pam-pld-0.75.6/modules/pam_debug/Makefile.am	Thu Jun 13 14:14:05 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_debug.la
+pkglib_LTLIBRARIES	= pam_debug.la
 endif
 
 pam_debug_la_SOURCES	= pam_debug.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_debug.lo
-libpam_debug_lo_SOURCES	= pam_debug.c
 libpam_debug.lo$(EXEEXT): pam_debug.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_deny/Makefile.am pam-pld-0.75.6/modules/pam_deny/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_deny/Makefile.am	Fri Mar  1 01:46:21 2002
+++ pam-pld-0.75.6/modules/pam_deny/Makefile.am	Thu Jun 13 14:14:10 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_deny.la
+pkglib_LTLIBRARIES	= pam_deny.la
 endif
 
 pam_deny_la_SOURCES	= pam_deny.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_deny.lo
-libpam_deny_lo_SOURCES	= pam_deny.c
 libpam_deny.lo$(EXEEXT): pam_deny.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_env/Makefile.am pam-pld-0.75.6/modules/pam_env/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_env/Makefile.am	Fri Mar  1 01:46:21 2002
+++ pam-pld-0.75.6/modules/pam_env/Makefile.am	Thu Jun 13 14:14:16 2002
@@ -8,7 +8,7 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 env_confdir		= /etc/security
 env_conf_DATA		= pam_env.conf
@@ -20,7 +20,7 @@
 			  -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_env.la
+pkglib_LTLIBRARIES	= pam_env.la
 endif
 
 pam_env_la_SOURCES	= pam_env.c
@@ -39,7 +39,6 @@
 			  -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_env.lo
-libpam_env_lo_SOURCES	= pam_env.c
 libpam_env.lo$(EXEEXT): pam_env.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_filter/Makefile.am pam-pld-0.75.6/modules/pam_filter/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_filter/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_filter/Makefile.am	Thu Jun 13 14:14:23 2002
@@ -10,12 +10,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_filter.la
+pkglib_LTLIBRARIES	= pam_filter.la
 endif
 
 pam_filter_la_SOURCES	= pam_filter.c
@@ -35,7 +35,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_filter.lo
-libpam_filter_lo_SOURCES	= pam_filter.c
 libpam_filter.lo$(EXEEXT): pam_filter.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_ftp/Makefile.am pam-pld-0.75.6/modules/pam_ftp/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_ftp/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_ftp/Makefile.am	Thu Jun 13 14:14:34 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_ftp.la
+pkglib_LTLIBRARIES	= pam_ftp.la
 endif
 
 pam_ftp_la_SOURCES	= pam_ftp.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_ftp.lo
-libpam_ftp_lo_SOURCES	= pam_ftp.c
 libpam_ftp.lo$(EXEEXT): pam_ftp.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_group/Makefile.am pam-pld-0.75.6/modules/pam_group/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_group/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_group/Makefile.am	Thu Jun 13 14:14:40 2002
@@ -8,7 +8,7 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 group_confdir		= /etc/security
 group_conf_DATA		= group.conf
@@ -17,7 +17,7 @@
 			  -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_group.la
+pkglib_LTLIBRARIES	= pam_group.la
 endif
 
 pam_group_la_SOURCES	= pam_group.c
@@ -39,7 +39,6 @@
 			  -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_group.lo
-libpam_group_lo_SOURCES	= pam_group.c
 libpam_group.lo$(EXEEXT): pam_group.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_homedir/Makefile.am pam-pld-0.75.6/modules/pam_homedir/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_homedir/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_homedir/Makefile.am	Thu Jun 13 14:14:45 2002
@@ -7,12 +7,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_homedir.la
+pkglib_LTLIBRARIES	= pam_homedir.la
 endif
 
 pam_homedir_la_SOURCES	= pam_homedir.c
@@ -28,7 +28,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_homedir.lo
-libpam_homedir_lo_SOURCES	= pam_homedir.c
 libpam_homedir.lo$(EXEEXT): pam_homedir.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_issue/Makefile.am pam-pld-0.75.6/modules/pam_issue/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_issue/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_issue/Makefile.am	Thu Jun 13 14:14:50 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_issue.la
+pkglib_LTLIBRARIES	= pam_issue.la
 endif
 
 pam_issue_la_SOURCES	= pam_issue.c
@@ -28,7 +28,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_issue.lo
-libpam_issue_lo_SOURCES	= pam_issue.c
 libpam_issue.lo$(EXEEXT): pam_issue.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_lastlog/Makefile.am pam-pld-0.75.6/modules/pam_lastlog/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_lastlog/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_lastlog/Makefile.am	Thu Jun 13 14:14:54 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_lastlog.la
+pkglib_LTLIBRARIES	= pam_lastlog.la
 endif
 
 pam_lastlog_la_SOURCES	= pam_lastlog.c
@@ -32,7 +32,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_lastlog.lo
-libpam_lastlog_lo_SOURCES	= pam_lastlog.c
 libpam_lastlog.lo$(EXEEXT): pam_lastlog.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_limits/Makefile.am pam-pld-0.75.6/modules/pam_limits/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_limits/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_limits/Makefile.am	Thu Jun 13 14:15:08 2002
@@ -8,7 +8,7 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 limits_confdir		= /etc/security
 limits_conf_DATA	= limits.conf
@@ -20,7 +20,7 @@
 			  -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_limits.la
+pkglib_LTLIBRARIES	= pam_limits.la
 endif
 
 pam_limits_la_SOURCES	= pam_limits.c
@@ -39,7 +39,6 @@
 			  -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_limits.lo
-libpam_limits_lo_SOURCES	= pam_limits.c
 libpam_limits.lo$(EXEEXT): pam_limits.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_listfile/Makefile.am pam-pld-0.75.6/modules/pam_listfile/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_listfile/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_listfile/Makefile.am	Thu Jun 13 14:15:14 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_listfile.la
+pkglib_LTLIBRARIES	= pam_listfile.la
 endif
 
 pam_listfile_la_SOURCES	= pam_listfile.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_listfile.lo
-libpam_listfile_lo_SOURCES	= pam_listfile.c
 libpam_listfile.lo$(EXEEXT): pam_listfile.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_mail/Makefile.am pam-pld-0.75.6/modules/pam_mail/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_mail/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_mail/Makefile.am	Thu Jun 13 14:15:17 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_mail.la
+pkglib_LTLIBRARIES	= pam_mail.la
 endif
 
 pam_mail_la_SOURCES	= pam_mail.c
@@ -34,7 +34,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_mail.lo
-libpam_mail_lo_SOURCES	= pam_mail.c
 libpam_mail.lo$(EXEEXT): pam_mail.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_make/Makefile.am pam-pld-0.75.6/modules/pam_make/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_make/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_make/Makefile.am	Thu Jun 13 14:15:21 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_make.la
+pkglib_LTLIBRARIES	= pam_make.la
 endif
 
 pam_make_la_SOURCES	= pam_make.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_make.lo
-libpam_make_lo_SOURCES	= pam_make.c
 libpam_make.lo$(EXEEXT): pam_make.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_motd/Makefile.am pam-pld-0.75.6/modules/pam_motd/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_motd/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_motd/Makefile.am	Thu Jun 13 14:15:25 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_motd.la
+pkglib_LTLIBRARIES	= pam_motd.la
 endif
 
 pam_motd_la_SOURCES	= pam_motd.c
@@ -29,7 +29,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_motd.lo
-libpam_motd_lo_SOURCES	= pam_motd.c
 libpam_motd.lo$(EXEEXT): pam_motd.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_netid/Makefile.am pam-pld-0.75.6/modules/pam_netid/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_netid/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_netid/Makefile.am	Thu Jun 13 14:15:28 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_netid.la
+pkglib_LTLIBRARIES	= pam_netid.la
 endif
 
 pam_netid_la_SOURCES	= pam_netid.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_netid.lo
-libpam_netid_lo_SOURCES	= pam_netid.c
 libpam_netid.lo$(EXEEXT): pam_netid.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_nologin/Makefile.am pam-pld-0.75.6/modules/pam_nologin/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_nologin/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_nologin/Makefile.am	Thu Jun 13 14:15:32 2002
@@ -8,12 +8,12 @@
 
 LIBS			= -lc
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_nologin.la
+pkglib_LTLIBRARIES	= pam_nologin.la
 endif
 
 pam_nologin_la_SOURCES	= pam_nologin.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_nologin.lo
-libpam_nologin_lo_SOURCES	= pam_nologin.c
 libpam_nologin.lo$(EXEEXT): pam_nologin.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_opie/Makefile.am pam-pld-0.75.6/modules/pam_opie/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_opie/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_opie/Makefile.am	Thu Jun 13 14:16:24 2002
@@ -7,7 +7,7 @@
 if OPIELIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_opie.la pam_opietrust.la
+pkglib_LTLIBRARIES	= pam_opie.la pam_opietrust.la
 endif
 
 pam_opie_la_SOURCES	= pam_opie.c
@@ -30,7 +30,7 @@
 pam_opietrust_la_LIBADD	= -lopie
 endif
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
@@ -48,8 +48,6 @@
 
 if OPIELIB_PRESENT
 noinst_PROGRAMS		= libpam_opie.lo libpam_opietrust.lo
-libpam_opie_lo_SOURCES	= pam_opie.c
-libpam_opietrust_lo_SOURCES	= pam_opietrust.c
 endif
 
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
diff -Nur pam-pld-0.75.6.orig/modules/pam_permit/Makefile.am pam-pld-0.75.6/modules/pam_permit/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_permit/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_permit/Makefile.am	Thu Jun 13 14:16:27 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_permit.la
+pkglib_LTLIBRARIES	= pam_permit.la
 endif
 
 pam_permit_la_SOURCES	= pam_permit.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_permit.lo
-libpam_permit_lo_SOURCES	= pam_permit.c
 libpam_permit.lo$(EXEEXT): pam_permit.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_pwdb/Makefile.am pam-pld-0.75.6/modules/pam_pwdb/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_pwdb/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_pwdb/Makefile.am	Thu Jun 13 14:16:35 2002
@@ -7,7 +7,7 @@
 if PWDBLIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_pwdb.la
+pkglib_LTLIBRARIES	= pam_pwdb.la
 endif
 
 pam_pwdb_la_SOURCES	= pam_pwdb.c
@@ -25,7 +25,7 @@
 
 LIBS			= -lpwdb
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 sbindir			= /sbin
 
 INCLUDES		= -DCHKPWD_HELPER=\"$(sbindir)/$(sbin_PROGRAMS)\" \
@@ -52,7 +52,6 @@
 if PWDBLIB_PRESENT
 
 noinst_PROGRAMS		= libpam_pwdb.lo
-libpam_pwdb_lo_SOURCES	= $(pam_pwdb_la_SOURCES)
 
 endif
 
diff -Nur pam-pld-0.75.6.orig/modules/pam_pwexport/Makefile.am pam-pld-0.75.6/modules/pam_pwexport/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_pwexport/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_pwexport/Makefile.am	Thu Jun 13 14:16:47 2002
@@ -13,12 +13,12 @@
 
 TITLE			= pam_pwexport
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_pwexport.la
+pkglib_LTLIBRARIES	= pam_pwexport.la
 endif
 
 CLEANFILES		= *~ $(TITLE).so
@@ -30,7 +30,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_pwexport.lo
-libpam_pwexport_lo_SOURCES	= pam_pwexport.c
 libpam_pwexport.lo$(EXEEXT): pam_pwexport.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_pwgen/Makefile.am pam-pld-0.75.6/modules/pam_pwgen/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_pwgen/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_pwgen/Makefile.am	Thu Jun 13 14:16:52 2002
@@ -8,7 +8,7 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 pwgen_confdir		= /etc/security
 pwgen_conf_DATA		= trigram.de trigram.dk trigram.en trigram.es \
@@ -19,7 +19,7 @@
 			  -DDEFAULT_TRIGRAM_FILE=\"$(pwgen_confdir)/trigram\"
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_pwgen.la
+pkglib_LTLIBRARIES	= pam_pwgen.la
 endif
 
 pam_pwgen_la_SOURCES	= pam_pwgen.c
@@ -47,7 +47,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_pwgen.lo
-libpam_pwgen_lo_SOURCES	= pam_pwgen.c
 libpam_pwgen.lo$(EXEEXT): pam_pwgen.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_radius/Makefile.am pam-pld-0.75.6/modules/pam_radius/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_radius/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_radius/Makefile.am	Thu Jun 13 14:16:57 2002
@@ -7,7 +7,7 @@
 if PWDBLIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_radius.la
+pkglib_LTLIBRARIES	= pam_radius.la
 endif
 
 pam_radius_la_SOURCES	= pam_radius.c
@@ -20,7 +20,7 @@
 
 LIBS			= -lpwdb
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 radius_confdir		= /etc/security
 radius_conf		= radius.conf
@@ -44,7 +44,6 @@
 if PWDBLIB_PRESENT
 
 noinst_PROGRAMS		= libpam_radius.lo
-libpam_radius_lo_SOURCES	= pam_radius.c
 
 endif
 
diff -Nur pam-pld-0.75.6.orig/modules/pam_rhosts/Makefile.am pam-pld-0.75.6/modules/pam_rhosts/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_rhosts/Makefile.am	Wed Mar  6 23:54:35 2002
+++ pam-pld-0.75.6/modules/pam_rhosts/Makefile.am	Thu Jun 13 14:17:01 2002
@@ -8,13 +8,13 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include \
 			  -I$(top_srcdir)/modules/pammodutil/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_rhosts.la
+pkglib_LTLIBRARIES	= pam_rhosts.la
 endif
 
 pam_rhosts_la_SOURCES	= pam_rhosts_auth.c
@@ -32,7 +32,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_rhosts.lo
-libpam_rhosts_lo_SOURCES	= pam_rhosts_auth.c
 libpam_rhosts.lo$(EXEEXT): pam_rhosts_auth.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE)_auth.c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_rootok/Makefile.am pam-pld-0.75.6/modules/pam_rootok/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_rootok/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_rootok/Makefile.am	Thu Jun 13 14:17:06 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_rootok.la
+pkglib_LTLIBRARIES	= pam_rootok.la
 endif
 
 pam_rootok_la_SOURCES	= pam_rootok.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_rootok.lo
-libpam_rootok_lo_SOURCES	= pam_rootok.c
 libpam_rootok.lo$(EXEEXT): pam_rootok.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_securetty/Makefile.am pam-pld-0.75.6/modules/pam_securetty/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_securetty/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_securetty/Makefile.am	Thu Jun 13 14:17:11 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_securetty.la
+pkglib_LTLIBRARIES	= pam_securetty.la
 endif
 
 pam_securetty_la_SOURCES	= pam_securetty.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_securetty.lo
-libpam_securetty_lo_SOURCES	= pam_securetty.c
 libpam_securetty.lo$(EXEEXT): pam_securetty.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_shells/Makefile.am pam-pld-0.75.6/modules/pam_shells/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_shells/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_shells/Makefile.am	Thu Jun 13 14:17:15 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_shells.la
+pkglib_LTLIBRARIES	= pam_shells.la
 endif
 
 pam_shells_la_SOURCES	= pam_shells.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_shells.lo
-libpam_shells_lo_SOURCES	= pam_shells.c
 libpam_shells.lo$(EXEEXT): pam_shells.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_skey/Makefile.am pam-pld-0.75.6/modules/pam_skey/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_skey/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_skey/Makefile.am	Thu Jun 13 14:17:21 2002
@@ -7,7 +7,7 @@
 if SKEYLIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_skey.la
+pkglib_LTLIBRARIES	= pam_skey.la
 endif
 
 pam_skey_la_SOURCES	= pam_skey.c
@@ -26,7 +26,7 @@
 pam_skey_la_LIBADD	= -lskey
 endif
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
@@ -41,7 +41,6 @@
 
 if SKEYLIB_PRESENT
 noinst_PROGRAMS		= libpam_skey.lo
-libpam_skey_lo_SOURCES	= pam_skey.c
 endif
 
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
diff -Nur pam-pld-0.75.6.orig/modules/pam_stress/Makefile.am pam-pld-0.75.6/modules/pam_stress/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_stress/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_stress/Makefile.am	Thu Jun 13 14:17:27 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_stress.la
+pkglib_LTLIBRARIES	= pam_stress.la
 endif
 
 pam_stress_la_SOURCES	= pam_stress.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_stress.lo
-libpam_stress_lo_SOURCES	= pam_stress.c
 libpam_stress.lo$(EXEEXT): pam_stress.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_tally/Makefile.am pam-pld-0.75.6/modules/pam_tally/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_tally/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_tally/Makefile.am	Thu Jun 13 14:17:31 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_tally.la
+pkglib_LTLIBRARIES	= pam_tally.la
 endif
 
 pam_tally_la_SOURCES	= pam_tally.c
@@ -35,7 +35,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_tally.lo
-libpam_tally_lo_SOURCES	= pam_tally.c
 libpam_tally.lo$(EXEEXT): pam_tally.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_tcpd/Makefile.am pam-pld-0.75.6/modules/pam_tcpd/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_tcpd/Makefile.am	Fri Mar  1 01:46:22 2002
+++ pam-pld-0.75.6/modules/pam_tcpd/Makefile.am	Thu Jun 13 14:17:36 2002
@@ -7,7 +7,7 @@
 if WRAPLIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_tcpd.la
+pkglib_LTLIBRARIES	= pam_tcpd.la
 endif
 
 pam_tcpd_la_SOURCES	= pam_tcpd.c
@@ -22,7 +22,7 @@
 
 LIBS			= -lwrap
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 tcpd_confdir		= /etc/security
 
@@ -40,7 +40,6 @@
 
 if WRAPLIB_PRESENT
 noinst_PROGRAMS		= libpam_tcpd.lo
-libpam_tcpd_lo_SOURCES	= pam_tcpd.c
 endif
 
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
diff -Nur pam-pld-0.75.6.orig/modules/pam_time/Makefile.am pam-pld-0.75.6/modules/pam_time/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_time/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_time/Makefile.am	Thu Jun 13 14:17:40 2002
@@ -8,7 +8,7 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 time_confdir		= /etc/security
 time_conf_DATA		= time.conf
@@ -17,7 +17,7 @@
 			  -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_time.la
+pkglib_LTLIBRARIES	= pam_time.la
 endif
 
 pam_time_la_SOURCES	= pam_time.c
@@ -36,7 +36,6 @@
 			  -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_time.lo
-libpam_time_lo_SOURCES	= pam_time.c
 libpam_time.lo$(EXEEXT): pam_time.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_unix/Makefile.am pam-pld-0.75.6/modules/pam_unix/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_unix/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_unix/Makefile.am	Thu Jun 13 14:17:47 2002
@@ -8,7 +8,7 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 sbindir			= /sbin
 
 INCLUDES		= -DCHKPWD_HELPER=\"$(sbindir)/$(sbin_PROGRAMS)\" \
@@ -16,7 +16,7 @@
 			  -I../../pamcrypt/include
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_unix.la
+pkglib_LTLIBRARIES	= pam_unix.la
 endif
 
 pam_unix_la_SOURCES	= pam_unix_acct.c pam_unix_auth.c \
@@ -58,7 +58,6 @@
 			  -I../../pamcrypt/include
 			   
 noinst_PROGRAMS		= libpam_unix.lo
-libpam_unix_lo_SOURCES	= $(pam_unix_la_SOURCES)
 libpam_unix_OBJECTS	= static/pam_unix_acct.lo \
 			  static/pam_unix_auth.lo \
 			  static/pam_unix_passwd.lo \
diff -Nur pam-pld-0.75.6.orig/modules/pam_userdb/Makefile.am pam-pld-0.75.6/modules/pam_userdb/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_userdb/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_userdb/Makefile.am	Thu Jun 13 14:17:52 2002
@@ -7,7 +7,7 @@
 if DBLIB_PRESENT
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_userdb.la
+pkglib_LTLIBRARIES	= pam_userdb.la
 endif
 
 pam_userdb_la_SOURCES	= pam_userdb.c
@@ -22,7 +22,7 @@
 
 LIBS			= -ldb
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 userdb_confdir		= /etc/security
 userdb_conf		= userdb.conf
@@ -48,7 +48,6 @@
 if DBLIB_PRESENT
 
 noinst_PROGRAMS		= libpam_userdb.lo
-libpam_userdb_lo_SOURCES	= $(pam_userdb_la_SOURCES)
 
 endif
 
diff -Nur pam-pld-0.75.6.orig/modules/pam_usertty/Makefile.am pam-pld-0.75.6/modules/pam_usertty/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_usertty/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_usertty/Makefile.am	Thu Jun 13 14:17:57 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_usertty.la
+pkglib_LTLIBRARIES	= pam_usertty.la
 endif
 
 pam_usertty_la_SOURCES	= pam_usertty.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_usertty.lo
-libpam_usertty_lo_SOURCES	= pam_usertty.c
 libpam_usertty.lo$(EXEEXT): pam_usertty.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_utmp/Makefile.am pam-pld-0.75.6/modules/pam_utmp/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_utmp/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_utmp/Makefile.am	Thu Jun 13 14:18:01 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_utmp.la
+pkglib_LTLIBRARIES	= pam_utmp.la
 endif
 
 pam_utmp_la_SOURCES	= pam_utmp.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_utmp.lo
-libpam_utmp_lo_SOURCES	= pam_utmp.c
 libpam_utmp.lo$(EXEEXT): pam_utmp.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_warn/Makefile.am pam-pld-0.75.6/modules/pam_warn/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_warn/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_warn/Makefile.am	Thu Jun 13 14:18:04 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_warn.la
+pkglib_LTLIBRARIES	= pam_warn.la
 endif
 
 pam_warn_la_SOURCES	= pam_warn.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_warn.lo
-libpam_warn_lo_SOURCES	= pam_warn.c
 libpam_warn.lo$(EXEEXT): pam_warn.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_wheel/Makefile.am pam-pld-0.75.6/modules/pam_wheel/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_wheel/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_wheel/Makefile.am	Thu Jun 13 14:18:07 2002
@@ -8,12 +8,12 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_wheel.la
+pkglib_LTLIBRARIES	= pam_wheel.la
 endif
 
 pam_wheel_la_SOURCES	= pam_wheel.c
@@ -31,7 +31,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_wheel.lo
-libpam_wheel_lo_SOURCES	= pam_wheel.c
 libpam_wheel.lo$(EXEEXT): pam_wheel.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
diff -Nur pam-pld-0.75.6.orig/modules/pam_xauth/Makefile.am pam-pld-0.75.6/modules/pam_xauth/Makefile.am
--- pam-pld-0.75.6.orig/modules/pam_xauth/Makefile.am	Fri Mar  1 01:46:23 2002
+++ pam-pld-0.75.6/modules/pam_xauth/Makefile.am	Thu Jun 13 14:18:15 2002
@@ -8,14 +8,14 @@
 
 LIBS			=
 
-libexecdir		= /lib/security
+pkglibdir		= /lib/security
 
 man_MANS		= pam_xauth.8
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 if DYNAMIC_MODULES
-libexec_LTLIBRARIES	= pam_xauth.la
+pkglib_LTLIBRARIES	= pam_xauth.la
 endif
 
 pam_xauth_la_SOURCES	= pam_xauth.c
@@ -33,7 +33,6 @@
 STATIC_INCLUDES		= -I. -I../.. -I$(top_srcdir)/include $(STATIC)
 			   
 noinst_PROGRAMS		= libpam_xauth.lo
-libpam_xauth_lo_SOURCES	= pam_xauth.c
 libpam_xauth.lo$(EXEEXT): pam_xauth.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
