Index: pam/modules/pam_userdb/pam_userdb.c
===================================================================
RCS file: /cvsroot/pam/modules/pam_userdb/pam_userdb.c,v
retrieving revision 1.10
retrieving revision 1.11
diff -u -r1.10 -r1.11
--- pam/modules/pam_userdb/pam_userdb.c	7 Apr 2005 12:25:50 -0000	1.10
+++ pam/modules/pam_userdb/pam_userdb.c	1 Jun 2005 00:22:33 -0000	1.11
@@ -1,7 +1,7 @@
 /* pam_userdb module */
  
 /*
- * $Id$
+ * $Id$
  * Written by Cristian Gafton <gafton@redhat.com> 1996/09/10
  * See the end of the file for Copyright Information
  */
@@ -309,6 +309,9 @@
         return PAM_SERVICE_ERR;
      }
 
+     if (cryptmode == NULL)
+	cryptmode = strdup("none");
+
      /* Get the username */
      retval = pam_get_user(pamh, &username, NULL);
      if ((retval != PAM_SUCCESS) || (!username)) {
@@ -402,6 +405,14 @@
 
     /* parse arguments */
     ctrl = _pam_parse(argc, argv, &database, &cryptmode);
+    if ((database == NULL) || (strlen(database) == 0)) {
+        if (ctrl & PAM_DEBUG_ARG)
+            _pam_log(LOG_DEBUG,"can not get the database name");
+        return PAM_SERVICE_ERR;
+    }
+
+    if (cryptmode == NULL)
+	cryptmode = strdup("none");
 
     /* Get the username */
     retval = pam_get_user(pamh, &username, NULL);
