diff -Nur pam-pld-0.74.3.orig/modules/pam_access/Makefile.am pam-pld-0.74.3/modules/pam_access/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_access/Makefile.am	Mon Dec  4 16:50:44 2000
+++ pam-pld-0.74.3/modules/pam_access/Makefile.am	Wed Oct 31 23:54:34 2001
@@ -37,7 +37,7 @@
 			   
 noinst_PROGRAMS		= libpam_access.lo
 libpam_access_lo_SOURCES = pam_access.c
-libpam_access.lo: pam_access.c
+libpam_access.lo$(EXEEXT): pam_access.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_cap/Makefile.am pam-pld-0.74.3/modules/pam_cap/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_cap/Makefile.am	Tue Apr 25 15:03:08 2000
+++ pam-pld-0.74.3/modules/pam_cap/Makefile.am	Wed Oct 31 23:54:49 2001
@@ -50,7 +50,7 @@
 STATIC_INCLUDES		= -DCONFILE=\"$(cap_confdir)/$(cap_conf_DATA)\" \
 			  -I$(top_srcdir)/include $(STATIC)
 			   
-libpam_cap.lo: pam_cap.c
+libpam_cap.lo$(EXEEXT): pam_cap.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_console/Makefile.am pam-pld-0.74.3/modules/pam_console/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_console/Makefile.am	Mon Dec  6 20:06:16 1999
+++ pam-pld-0.74.3/modules/pam_console/Makefile.am	Wed Oct 31 23:55:01 2001
@@ -38,7 +38,7 @@
 			   
 noinst_PROGRAMS		= libpam_console.lo
 libpam_console_lo_SOURCES = pam_console.c
-libpam_console.lo: pam_console.c
+libpam_console.lo$(EXEEXT): pam_console.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_cracklib/Makefile.am pam-pld-0.74.3/modules/pam_cracklib/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_cracklib/Makefile.am	Mon Nov 15 20:12:16 1999
+++ pam-pld-0.74.3/modules/pam_cracklib/Makefile.am	Wed Oct 31 23:59:30 2001
@@ -18,19 +18,17 @@
 
 TITLE			= pam_cracklib
 
-if CRYPTLIB_PRESENT
-LIBS			= -lcrypt -lcrack
-else
-LIBS			= -lcrack
-endif
-
 libexecdir		= /lib/security
 
 INCLUDES		= -DCRACKLIB_DICTPATH=\"$(CRACKLIB_DICTPATH)\" \
 			  -I$(top_srcdir)/include $(DYNAMIC)
 
 pam_cracklib_la_LDFLAGS	= -avoid-version -module
-pam_cracklib_la_LIBADD	=
+if CRYPTLIB_PRESENT
+pam_cracklib_la_LIBADD	= -lcrypt -lcrack
+else
+pam_cracklib_la_LIBADD	= -lcrack
+endif
 
 EXTRA_DIST		= README
 
@@ -50,7 +48,7 @@
 STATIC_INCLUDES		= -DCRACKLIB_DICTPATH=\"$(CRACKLIB_DICTPATH)\"
 			  -I$(top_srcdir)/include $(STATIC)
 			   
-libpam_cracklib.lo: pam_cracklib.c
+libpam_cracklib.lo$(EXEEXT): pam_cracklib.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_deny/Makefile.am pam-pld-0.74.3/modules/pam_deny/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_deny/Makefile.am	Mon Nov 15 20:12:17 1999
+++ pam-pld-0.74.3/modules/pam_deny/Makefile.am	Wed Oct 31 23:59:45 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_deny.lo
 libpam_deny_lo_SOURCES	= pam_deny.c
-libpam_deny.lo: pam_deny.c
+libpam_deny.lo$(EXEEXT): pam_deny.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_env/Makefile.am pam-pld-0.74.3/modules/pam_env/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_env/Makefile.am	Mon Dec  4 15:43:32 2000
+++ pam-pld-0.74.3/modules/pam_env/Makefile.am	Wed Oct 31 23:59:56 2001
@@ -40,7 +40,7 @@
 			   
 noinst_PROGRAMS		= libpam_env.lo
 libpam_env_lo_SOURCES	= pam_env.c
-libpam_env.lo: pam_env.c
+libpam_env.lo$(EXEEXT): pam_env.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_filter/Makefile.am pam-pld-0.74.3/modules/pam_filter/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_filter/Makefile.am	Mon Nov 15 20:12:17 1999
+++ pam-pld-0.74.3/modules/pam_filter/Makefile.am	Thu Nov  1 00:00:07 2001
@@ -36,7 +36,7 @@
 			   
 noinst_PROGRAMS		= libpam_filter.lo
 libpam_filter_lo_SOURCES	= pam_filter.c
-libpam_filter.lo: pam_filter.c
+libpam_filter.lo$(EXEEXT): pam_filter.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_ftp/Makefile.am pam-pld-0.74.3/modules/pam_ftp/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_ftp/Makefile.am	Mon Nov 15 20:12:20 1999
+++ pam-pld-0.74.3/modules/pam_ftp/Makefile.am	Thu Nov  1 00:00:18 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_ftp.lo
 libpam_ftp_lo_SOURCES	= pam_ftp.c
-libpam_ftp.lo: pam_ftp.c
+libpam_ftp.lo$(EXEEXT): pam_ftp.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_group/Makefile.am pam-pld-0.74.3/modules/pam_group/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_group/Makefile.am	Mon Dec  4 16:50:44 2000
+++ pam-pld-0.74.3/modules/pam_group/Makefile.am	Thu Nov  1 00:00:31 2001
@@ -40,7 +40,7 @@
 			   
 noinst_PROGRAMS		= libpam_group.lo
 libpam_group_lo_SOURCES	= pam_group.c
-libpam_group.lo: pam_group.c
+libpam_group.lo$(EXEEXT): pam_group.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_homedir/Makefile.am pam-pld-0.74.3/modules/pam_homedir/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_homedir/Makefile.am	Mon Nov 15 20:12:21 1999
+++ pam-pld-0.74.3/modules/pam_homedir/Makefile.am	Thu Nov  1 00:00:41 2001
@@ -29,7 +29,7 @@
 			   
 noinst_PROGRAMS		= libpam_homedir.lo
 libpam_homedir_lo_SOURCES	= pam_homedir.c
-libpam_homedir.lo: pam_homedir.c
+libpam_homedir.lo$(EXEEXT): pam_homedir.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_issue/Makefile.am pam-pld-0.74.3/modules/pam_issue/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_issue/Makefile.am	Mon Nov 15 20:12:21 1999
+++ pam-pld-0.74.3/modules/pam_issue/Makefile.am	Thu Nov  1 00:00:52 2001
@@ -29,7 +29,7 @@
 			   
 noinst_PROGRAMS		= libpam_issue.lo
 libpam_issue_lo_SOURCES	= pam_issue.c
-libpam_issue.lo: pam_issue.c
+libpam_issue.lo$(EXEEXT): pam_issue.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_lastlog/Makefile.am pam-pld-0.74.3/modules/pam_lastlog/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_lastlog/Makefile.am	Mon Nov 15 20:12:22 1999
+++ pam-pld-0.74.3/modules/pam_lastlog/Makefile.am	Thu Nov  1 00:01:03 2001
@@ -33,7 +33,7 @@
 			   
 noinst_PROGRAMS		= libpam_lastlog.lo
 libpam_lastlog_lo_SOURCES	= pam_lastlog.c
-libpam_lastlog.lo: pam_lastlog.c
+libpam_lastlog.lo$(EXEEXT): pam_lastlog.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_limits/Makefile.am pam-pld-0.74.3/modules/pam_limits/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_limits/Makefile.am	Sun Oct  1 13:01:39 2000
+++ pam-pld-0.74.3/modules/pam_limits/Makefile.am	Thu Nov  1 00:01:13 2001
@@ -40,7 +40,7 @@
 			   
 noinst_PROGRAMS		= libpam_limits.lo
 libpam_limits_lo_SOURCES	= pam_limits.c
-libpam_limits.lo: pam_limits.c
+libpam_limits.lo$(EXEEXT): pam_limits.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_listfile/Makefile.am pam-pld-0.74.3/modules/pam_listfile/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_listfile/Makefile.am	Mon Nov 15 20:12:23 1999
+++ pam-pld-0.74.3/modules/pam_listfile/Makefile.am	Thu Nov  1 00:01:23 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_listfile.lo
 libpam_listfile_lo_SOURCES	= pam_listfile.c
-libpam_listfile.lo: pam_listfile.c
+libpam_listfile.lo$(EXEEXT): pam_listfile.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_mail/Makefile.am pam-pld-0.74.3/modules/pam_mail/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_mail/Makefile.am	Mon Nov 15 20:12:23 1999
+++ pam-pld-0.74.3/modules/pam_mail/Makefile.am	Thu Nov  1 00:01:32 2001
@@ -35,7 +35,7 @@
 			   
 noinst_PROGRAMS		= libpam_mail.lo
 libpam_mail_lo_SOURCES	= pam_mail.c
-libpam_mail.lo: pam_mail.c
+libpam_mail.lo$(EXEEXT): pam_mail.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_make/Makefile.am pam-pld-0.74.3/modules/pam_make/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_make/Makefile.am	Mon Nov 15 20:12:23 1999
+++ pam-pld-0.74.3/modules/pam_make/Makefile.am	Thu Nov  1 00:01:44 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_make.lo
 libpam_make_lo_SOURCES	= pam_make.c
-libpam_make.lo: pam_make.c
+libpam_make.lo$(EXEEXT): pam_make.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_motd/Makefile.am pam-pld-0.74.3/modules/pam_motd/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_motd/Makefile.am	Mon Nov 15 20:12:24 1999
+++ pam-pld-0.74.3/modules/pam_motd/Makefile.am	Thu Nov  1 00:01:53 2001
@@ -30,7 +30,7 @@
 			   
 noinst_PROGRAMS		= libpam_motd.lo
 libpam_motd_lo_SOURCES	= pam_motd.c
-libpam_motd.lo: pam_motd.c
+libpam_motd.lo$(EXEEXT): pam_motd.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_netid/Makefile.am pam-pld-0.74.3/modules/pam_netid/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_netid/Makefile.am	Tue Apr 25 15:03:30 2000
+++ pam-pld-0.74.3/modules/pam_netid/Makefile.am	Thu Nov  1 00:02:08 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_netid.lo
 libpam_netid_lo_SOURCES	= pam_netid.c
-libpam_netid.lo: pam_netid.c
+libpam_netid.lo$(EXEEXT): pam_netid.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_nologin/Makefile.am pam-pld-0.74.3/modules/pam_nologin/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_nologin/Makefile.am	Mon Nov 15 20:12:24 1999
+++ pam-pld-0.74.3/modules/pam_nologin/Makefile.am	Thu Nov  1 00:02:31 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_nologin.lo
 libpam_nologin_lo_SOURCES	= pam_nologin.c
-libpam_nologin.lo: pam_nologin.c
+libpam_nologin.lo$(EXEEXT): pam_nologin.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_opie/Makefile.am pam-pld-0.74.3/modules/pam_opie/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_opie/Makefile.am	Tue Apr 25 15:04:01 2000
+++ pam-pld-0.74.3/modules/pam_opie/Makefile.am	Thu Nov  1 00:04:02 2001
@@ -23,9 +23,11 @@
 TITLE			= pam_opie
 
 if CRYPTLIB_PRESENT
-LIBS			= -lcrypt -lopie
+pam_opie_la_LIBADD	= -lcrypt -lopie
+pam_opietrust_la_LIBADD	= -lcrypt -lopie
 else
-LIBS			= -lopie
+pam_opie_la_LIBADD	= -lopie
+pam_opietrust_la_LIBADD	= -lopie
 endif
 
 libexecdir		= /lib/security
@@ -33,9 +35,7 @@
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 pam_opie_la_LDFLAGS	= -avoid-version -module
-pam_opie_la_LIBADD	=
 pam_opietrust_la_LDFLAGS= -avoid-version -module
-pam_opietrust_la_LIBADD	=
 
 EXTRA_DIST		= opie.conf
 
@@ -54,14 +54,14 @@
 
 STATIC_INCLUDES		= -I$(top_srcdir)/include $(STATIC)
 			   
-libpam_opie.lo: pam_opie.c
+libpam_opie.lo$(EXEEXT): pam_opie.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
 	$(LN_S) lib$(TITLE).lo lib$(TITLE).o
 	( cd .. ; ./register_static $(TITLE) $(TITLE)/$(noinst_PROGRAMS) )
 
-libpam_opietrust.lo: pam_opietrust.c
+libpam_opietrust.lo$(EXEEXT): pam_opietrust.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE)trust.c -o static/$(TITLE)trust.lo
 	$(LD) -r -o lib$(TITLE)trust.lo static/$(TITLE)trust.lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_permit/Makefile.am pam-pld-0.74.3/modules/pam_permit/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_permit/Makefile.am	Mon Nov 15 20:12:24 1999
+++ pam-pld-0.74.3/modules/pam_permit/Makefile.am	Thu Nov  1 00:04:15 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_permit.lo
 libpam_permit_lo_SOURCES	= pam_permit.c
-libpam_permit.lo: pam_permit.c
+libpam_permit.lo$(EXEEXT): pam_permit.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_pwdb/Makefile.am pam-pld-0.74.3/modules/pam_pwdb/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_pwdb/Makefile.am	Mon Nov 15 20:12:25 1999
+++ pam-pld-0.74.3/modules/pam_pwdb/Makefile.am	Thu Nov  1 00:04:27 2001
@@ -60,7 +60,7 @@
 			  -DPWDB_HARD_CORE -I$(top_srcdir)/include $(STATIC) \
 			  -I../../pamcrypt/include
 			   
-libpam_pwdb.lo: pam_pwdb.c
+libpam_pwdb.lo$(EXEEXT): pam_pwdb.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo ../../pamcrypt/libpamcrypt.a static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_pwexport/Makefile.am pam-pld-0.74.3/modules/pam_pwexport/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_pwexport/Makefile.am	Wed Mar 28 00:03:42 2001
+++ pam-pld-0.74.3/modules/pam_pwexport/Makefile.am	Thu Nov  1 00:05:21 2001
@@ -13,8 +13,6 @@
 
 TITLE			= pam_pwexport
 
-LIBS			=
-
 libexecdir		= /lib/security
 
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
@@ -33,7 +31,7 @@
 			   
 noinst_PROGRAMS		= libpam_pwexport.lo
 libpam_pwexport_lo_SOURCES	= pam_pwexport.c
-libpam_pwexport.lo: pam_pwexport.c
+libpam_pwexport.lo$(EXEEXT): pam_pwexport.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_pwgen/Makefile.am pam-pld-0.74.3/modules/pam_pwgen/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_pwgen/Makefile.am	Mon Nov 15 20:12:27 1999
+++ pam-pld-0.74.3/modules/pam_pwgen/Makefile.am	Thu Nov  1 00:05:42 2001
@@ -48,7 +48,7 @@
 			   
 noinst_PROGRAMS		= libpam_pwgen.lo
 libpam_pwgen_lo_SOURCES	= pam_pwgen.c
-libpam_pwgen.lo: pam_pwgen.c
+libpam_pwgen.lo$(EXEEXT): pam_pwgen.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_radius/Makefile.am pam-pld-0.74.3/modules/pam_radius/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_radius/Makefile.am	Mon Nov 15 20:12:28 1999
+++ pam-pld-0.74.3/modules/pam_radius/Makefile.am	Thu Nov  1 00:06:01 2001
@@ -51,7 +51,7 @@
 STATIC_INCLUDES		= -DCONFILE=\"$(radius_confdir)/$(radius_conf)\" \
 			  -I$(top_srcdir)/include $(STATIC)
 			   
-libpam_radius.lo: pam_radius.c
+libpam_radius.lo$(EXEEXT): pam_radius.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_rhosts/Makefile.am pam-pld-0.74.3/modules/pam_rhosts/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_rhosts/Makefile.am	Mon Nov 15 20:12:28 1999
+++ pam-pld-0.74.3/modules/pam_rhosts/Makefile.am	Thu Nov  1 00:06:34 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_rhosts.lo
 libpam_rhosts_lo_SOURCES	= pam_rhosts_auth.c
-libpam_rhosts.lo: pam_rhosts_auth.c
+libpam_rhosts.lo$(EXEEXT): pam_rhosts_auth.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE)_auth.c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_rootok/Makefile.am pam-pld-0.74.3/modules/pam_rootok/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_rootok/Makefile.am	Mon Nov 15 20:12:28 1999
+++ pam-pld-0.74.3/modules/pam_rootok/Makefile.am	Thu Nov  1 00:06:45 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_rootok.lo
 libpam_rootok_lo_SOURCES	= pam_rootok.c
-libpam_rootok.lo: pam_rootok.c
+libpam_rootok.lo$(EXEEXT): pam_rootok.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_securetty/Makefile.am pam-pld-0.74.3/modules/pam_securetty/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_securetty/Makefile.am	Mon Nov 15 20:12:29 1999
+++ pam-pld-0.74.3/modules/pam_securetty/Makefile.am	Thu Nov  1 00:06:58 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_securetty.lo
 libpam_securetty_lo_SOURCES	= pam_securetty.c
-libpam_securetty.lo: pam_securetty.c
+libpam_securetty.lo$(EXEEXT): pam_securetty.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_shells/Makefile.am pam-pld-0.74.3/modules/pam_shells/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_shells/Makefile.am	Mon Nov 15 20:12:29 1999
+++ pam-pld-0.74.3/modules/pam_shells/Makefile.am	Thu Nov  1 00:07:12 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_shells.lo
 libpam_shells_lo_SOURCES	= pam_shells.c
-libpam_shells.lo: pam_shells.c
+libpam_shells.lo$(EXEEXT): pam_shells.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_skey/Makefile.am pam-pld-0.74.3/modules/pam_skey/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_skey/Makefile.am	Tue Apr 25 15:04:26 2000
+++ pam-pld-0.74.3/modules/pam_skey/Makefile.am	Thu Nov  1 00:08:18 2001
@@ -21,9 +21,9 @@
 TITLE			= pam_skey
 
 if CRYPTLIB_PRESENT
-LIBS			= -lcrypt -lskey
+pam_skey_la_LIBADD	= -lcrypt -lskey
 else
-LIBS			= -lskey
+pam_skey_la_LIBADD	= -lskey
 endif
 
 libexecdir		= /lib/security
@@ -31,7 +31,6 @@
 INCLUDES		= -I$(top_srcdir)/include $(DYNAMIC)
 
 pam_skey_la_LDFLAGS	= -avoid-version -module
-pam_skey_la_LIBADD	=
 
 CLEANFILES		= *~ $(TITLE).so
 
@@ -47,7 +46,7 @@
 
 STATIC_INCLUDES		= -I$(top_srcdir)/include $(STATIC)
 			   
-libpam_skey.lo: pam_skey.c
+libpam_skey.lo$(EXEEXT): pam_skey.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_stress/Makefile.am pam-pld-0.74.3/modules/pam_stress/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_stress/Makefile.am	Mon Nov 15 20:12:29 1999
+++ pam-pld-0.74.3/modules/pam_stress/Makefile.am	Thu Nov  1 00:08:33 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_stress.lo
 libpam_stress_lo_SOURCES	= pam_stress.c
-libpam_stress.lo: pam_stress.c
+libpam_stress.lo$(EXEEXT): pam_stress.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_tally/Makefile.am pam-pld-0.74.3/modules/pam_tally/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_tally/Makefile.am	Mon Nov 15 20:12:30 1999
+++ pam-pld-0.74.3/modules/pam_tally/Makefile.am	Thu Nov  1 00:08:43 2001
@@ -36,7 +36,7 @@
 			   
 noinst_PROGRAMS		= libpam_tally.lo
 libpam_tally_lo_SOURCES	= pam_tally.c
-libpam_tally.lo: pam_tally.c
+libpam_tally.lo$(EXEEXT): pam_tally.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_tcpd/Makefile.am pam-pld-0.74.3/modules/pam_tcpd/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_tcpd/Makefile.am	Tue Apr 25 15:04:59 2000
+++ pam-pld-0.74.3/modules/pam_tcpd/Makefile.am	Thu Nov  1 00:08:53 2001
@@ -45,7 +45,7 @@
 
 STATIC_INCLUDES		= -I$(top_srcdir)/include $(STATIC)
 			   
-libpam_tcpd.lo: pam_tcpd.c
+libpam_tcpd.lo$(EXEEXT): pam_tcpd.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_time/Makefile.am pam-pld-0.74.3/modules/pam_time/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_time/Makefile.am	Mon Dec  4 16:50:44 2000
+++ pam-pld-0.74.3/modules/pam_time/Makefile.am	Thu Nov  1 00:09:03 2001
@@ -37,7 +37,7 @@
 			   
 noinst_PROGRAMS		= libpam_time.lo
 libpam_time_lo_SOURCES	= pam_time.c
-libpam_time.lo: pam_time.c
+libpam_time.lo$(EXEEXT): pam_time.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_unix/Makefile.am pam-pld-0.74.3/modules/pam_unix/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_unix/Makefile.am	Mon Nov 15 20:12:31 1999
+++ pam-pld-0.74.3/modules/pam_unix/Makefile.am	Thu Nov  1 00:09:13 2001
@@ -65,7 +65,7 @@
 			  static/pam_unix_sess.lo \
 			  static/support.lo \
 			  static/yppasswd_xdr.lo
-libpam_unix.lo: $(libpam_unix_OBJECTS)
+libpam_unix.lo$(EXEEXT): $(libpam_unix_OBJECTS)
 	mkdir -p static
 	$(LD) -r -o lib$(TITLE).lo ../../pamcrypt/libpamcrypt.a $(libpam_unix_OBJECTS)
 	$(LN_S) lib$(TITLE).lo lib$(TITLE).o
diff -Nur pam-pld-0.74.3.orig/modules/pam_userdb/Makefile.am pam-pld-0.74.3/modules/pam_userdb/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_userdb/Makefile.am	Mon Nov 15 20:12:31 1999
+++ pam-pld-0.74.3/modules/pam_userdb/Makefile.am	Thu Nov  1 00:09:23 2001
@@ -55,7 +55,7 @@
 STATIC_INCLUDES		= -DCONFILE=\"$(userdb_confdir)/$(userdb_conf)\" \
 			  -I$(top_srcdir)/include $(STATIC)
 			   
-libpam_userdb.lo:
+libpam_userdb.lo$(EXEEXT):
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_usertty/Makefile.am pam-pld-0.74.3/modules/pam_usertty/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_usertty/Makefile.am	Tue Apr 25 15:05:28 2000
+++ pam-pld-0.74.3/modules/pam_usertty/Makefile.am	Thu Nov  1 00:09:33 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_usertty.lo
 libpam_usertty_lo_SOURCES	= pam_usertty.c
-libpam_usertty.lo: pam_usertty.c
+libpam_usertty.lo$(EXEEXT): pam_usertty.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_utmp/Makefile.am pam-pld-0.74.3/modules/pam_utmp/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_utmp/Makefile.am	Mon Nov 15 20:12:32 1999
+++ pam-pld-0.74.3/modules/pam_utmp/Makefile.am	Thu Nov  1 00:09:43 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_utmp.lo
 libpam_utmp_lo_SOURCES	= pam_utmp.c
-libpam_utmp.lo: pam_utmp.c
+libpam_utmp.lo$(EXEEXT): pam_utmp.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_warn/Makefile.am pam-pld-0.74.3/modules/pam_warn/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_warn/Makefile.am	Mon Nov 15 20:12:32 1999
+++ pam-pld-0.74.3/modules/pam_warn/Makefile.am	Thu Nov  1 00:09:56 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_warn.lo
 libpam_warn_lo_SOURCES	= pam_warn.c
-libpam_warn.lo: pam_warn.c
+libpam_warn.lo$(EXEEXT): pam_warn.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_wheel/Makefile.am pam-pld-0.74.3/modules/pam_wheel/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_wheel/Makefile.am	Mon Nov 15 20:12:34 1999
+++ pam-pld-0.74.3/modules/pam_wheel/Makefile.am	Thu Nov  1 00:10:05 2001
@@ -32,7 +32,7 @@
 			   
 noinst_PROGRAMS		= libpam_wheel.lo
 libpam_wheel_lo_SOURCES	= pam_wheel.c
-libpam_wheel.lo: pam_wheel.c
+libpam_wheel.lo$(EXEEXT): pam_wheel.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/modules/pam_xauth/Makefile.am pam-pld-0.74.3/modules/pam_xauth/Makefile.am
--- pam-pld-0.74.3.orig/modules/pam_xauth/Makefile.am	Mon Nov 15 20:12:35 1999
+++ pam-pld-0.74.3/modules/pam_xauth/Makefile.am	Thu Nov  1 00:10:35 2001
@@ -34,7 +34,7 @@
 			   
 noinst_PROGRAMS		= libpam_xauth.lo
 libpam_xauth_lo_SOURCES	= pam_xauth.c
-libpam_xauth.lo: pam_xauth.c
+libpam_xauth.lo$(EXEEXT): pam_xauth.c
 	mkdir -p static
 	$(LIBTOOL) --mode=compile $(CC) $(DEFS) $(STATIC_INCLUDES) $(CFLAGS) -c $(TITLE).c -o static/$(TITLE).lo
 	$(LD) -r -o lib$(TITLE).lo static/$(TITLE).lo
diff -Nur pam-pld-0.74.3.orig/pamcrypt/Makefile.am pam-pld-0.74.3/pamcrypt/Makefile.am
--- pam-pld-0.74.3.orig/pamcrypt/Makefile.am	Thu May 31 14:30:12 2001
+++ pam-pld-0.74.3/pamcrypt/Makefile.am	Thu Nov  1 00:11:32 2001
@@ -2,7 +2,9 @@
 noinst_LTLIBRARIES	= libpamcrypt.la
 
 if CRYPTLIB_PRESENT
-LIBS			= -lcrypt
+NEWLIBS			= -lcrypt
+else
+NEWLIBS			=
 endif
 
 if STRONG_CRYPTO
@@ -16,7 +18,7 @@
 
 libpamcrypt_la_LDFLAGS	=
 libpamcrypt_la_LIBADD	= md5_good.lo md5_broken.lo \
-			  md5_crypt_good.lo md5_crypt_broken.lo
+			  md5_crypt_good.lo md5_crypt_broken.lo $(NEWLIBS)
 
 if STRONG_CRYPTO
 noinst_HEADERS		= blf.h blowfish.h crypt16.h md5.h sha1.h include/pam_crypt.h
