--- Linux-PAM-0.99.9.0/modules/pam_tally/pam_tally.c~	2008-03-04 17:33:09.000000000 +0100
+++ Linux-PAM-0.99.9.0/modules/pam_tally/pam_tally.c	2008-03-04 17:44:16.000000000 +0100
@@ -504,6 +504,7 @@
 
     i=get_tally(pamh, &tally, uid, opts->filename, &TALLY, fsp);
     if ( i != PAM_SUCCESS ) { RETURN_ERROR( i ); }
+    fclose(TALLY);
 
     if ( !(opts->ctrl & OPT_MAGIC_ROOT) || getuid() ) {       /* magic_root skips tally check */
 
