--- Linux-PAM-1.1.6/modules/pam_namespace/Makefile.am~	2012-08-15 13:08:43.000000000 +0200
+++ Linux-PAM-1.1.6/modules/pam_namespace/Makefile.am	2012-11-27 12:11:03.796446562 +0100
@@ -40,7 +40,7 @@
   secureconf_SCRIPTS = namespace.init
 
 install-data-local:
-	mkdir -p $(namespaceddir)
+	mkdir -p $(DESTDIR)$(namespaceddir)
 endif
 
 
--- Linux-PAM-1.1.6/modules/pam_sepermit/Makefile.am~	2012-08-15 13:08:43.000000000 +0200
+++ Linux-PAM-1.1.6/modules/pam_sepermit/Makefile.am	2012-11-27 12:21:56.806423526 +0100
@@ -35,7 +35,7 @@
   securelib_LTLIBRARIES = pam_sepermit.la
 
 install-data-local:
-	mkdir -p $(sepermitlockdir)
+	mkdir -p $(DESTDIR)$(sepermitlockdir)
 endif
 if ENABLE_REGENERATE_MAN
 noinst_DATA = README pam_sepermit.8 sepermit.conf.5
