--- Linux-PAM-1.4.0/configure.ac.orig	2020-08-25 19:45:34.529916132 +0200
+++ Linux-PAM-1.4.0/configure.ac	2020-08-25 19:49:33.305289241 +0200
@@ -462,7 +462,7 @@
               LIBS=$old_libs
         fi
         if test -z "$LIBDB" ; then
-            AC_CHECK_LIB([ndbm],[dbm_store], LIBDB="-lndbm", LIBDB="")
+            AC_CHECK_LIB([gdbm],[dbm_store], LIBDB="-lgdbm -lgdbm_compat", LIBDB="", [-lgdbm_compat])
             if test -n "$LIBDB" ; then
                 AC_CHECK_HEADERS(ndbm.h)
             fi
