diff -Nur pam-pld-0.74.3.orig/modules/pam_limits/limits.skel pam-pld-0.74.3/modules/pam_limits/limits.skel
--- pam-pld-0.74.3.orig/modules/pam_limits/limits.skel	Mon Jul  5 16:34:40 1999
+++ pam-pld-0.74.3/modules/pam_limits/limits.skel	Fri Jun 15 19:44:05 2001
@@ -25,6 +25,7 @@
 #        - cpu - max CPU time (MIN)
 #        - nproc - max number of processes
 #        - as - address space limit
+#        - locks - file locks (requires kernel >= 2.4)
 #        - maxlogins - max number of logins for this user
 #        - priority - the priority to run user process with
 #
diff -Nur pam-pld-0.74.3.orig/modules/pam_limits/pam_limits.c pam-pld-0.74.3/modules/pam_limits/pam_limits.c
--- pam-pld-0.74.3.orig/modules/pam_limits/pam_limits.c	Mon Dec  4 15:18:44 2000
+++ pam-pld-0.74.3/modules/pam_limits/pam_limits.c	Fri Jun 15 19:40:38 2001
@@ -236,9 +236,12 @@
 
     D(("called."));
 
-    for(i = 0; i < RLIM_NLIMITS; i++) 
-	retval |= getrlimit(i, &pl->limits[i].limit);
-    
+    for(i = 0; i < RLIM_NLIMITS; i++)
+	if(i != RLIMIT_LOCKS)
+	    retval |= getrlimit(i, &pl->limits[i].limit);
+        else
+	    getrlimit(i, &pl->limits[i].limit);
+
     pl->limits[RLIMIT_CPU].src_soft = LIMITS_DEF_NONE;
     pl->limits[RLIMIT_CPU].src_hard = LIMITS_DEF_NONE;
 
@@ -269,6 +272,9 @@
     pl->limits[RLIMIT_AS].src_soft = LIMITS_DEF_NONE;
     pl->limits[RLIMIT_AS].src_hard = LIMITS_DEF_NONE;
 
+    pl->limits[RLIMIT_LOCKS].src_soft = LIMITS_DEF_NONE;
+    pl->limits[RLIMIT_LOCKS].src_hard = LIMITS_DEF_NONE;
+
     pl->priority = 0;
     pl->login_limit = -2;
     pl->login_limit_def = LIMITS_DEF_NONE;
@@ -310,6 +316,8 @@
 	limit_item = RLIMIT_MEMLOCK;
     else if (strcmp(lim_item, "as") == 0)
 	limit_item = RLIMIT_AS;
+    else if (strcmp(lim_item, "locks") == 0)
+	limit_item = RLIMIT_LOCKS;
     else if (strcmp(lim_item, "maxlogins") == 0) {
 	limit_item = LIMIT_LOGIN;
 	pl->flag_numsyslogins = 0;
@@ -498,7 +506,11 @@
     for (i=0; i<RLIM_NLIMITS; i++) {
         if (pl->limits[i].limit.rlim_cur > pl->limits[i].limit.rlim_max)
             pl->limits[i].limit.rlim_cur = pl->limits[i].limit.rlim_max;
-        retval |= setrlimit(i, &pl->limits[i].limit);
+	if(i != RLIMIT_LOCKS)
+	    retval |= setrlimit(i, &pl->limits[i].limit);
+	else if(setrlimit(i, &pl->limits[i].limit) &&
+	      (pl->limits[i].src_soft!=LIMITS_DEF_NONE || pl->limits[i].src_hard!=LIMITS_DEF_NONE))
+	    _pam_log(LOG_WARNING,"cannot set RLIMIT_LOCKS (kernel < 2.4?)");
     }
     
     if (retval != PAM_SUCCESS)
