diff -ur Linux-PAM-0.99.7.1/libpam/pam_modutil_private.h~ Linux-PAM-0.99.7.1/libpam/pam_modutil_private.h
--- Linux-PAM-0.99.7.1/libpam/pam_modutil_private.h~	2005-06-07 16:30:27.000000000 +0200
+++ Linux-PAM-0.99.7.1/libpam/pam_modutil_private.h	2006-06-07 08:48:49.000000000 +0200
@@ -12,7 +12,7 @@
 #include <security/pam_modutil.h>
 
 #define PWD_INITIAL_LENGTH     0x100
-#define PWD_ABSURD_PWD_LENGTH  0x8000
+#define PWD_ABSURD_PWD_LENGTH  0x20000
 
 extern void
 pammodutil_cleanup(pam_handle_t *pamh, void *data,
