--- pam-pld-0.78.1/modules/pam_mail/pam_mail.c.orig	2004-11-30 14:39:47.000000000 +0100
+++ pam-pld-0.78.1/modules/pam_mail/pam_mail.c	2004-12-23 10:21:18.000000000 +0100
@@ -282,7 +282,7 @@
 	    sprintf(dir, "%.250s/new", folder);
 	    i = scandir(dir, &namelist, 0, alphasort);
 	    if (i > 2) {
-		type = "new";
+		type = (ctrl & PAM_STANDARD_MAIL) ? "new " : "new";
 		while (--i)
 		    free(namelist[i]);
 	    } else {
@@ -291,7 +291,7 @@
 		sprintf(dir, "%.250s/cur", folder);
 		i = scandir(dir, &namelist, 0, alphasort);
 		if (i > 2) {
-		    type = "old";
+		    type = (ctrl & PAM_STANDARD_MAIL) ? "" : "old";
 		    while (--i)
 			free(namelist[i]);
 		} else if (ctrl & PAM_EMPTY_TOO) {
